### PR TITLE
add key and value types to VmInfoCard mapEntries

### PR DESCRIPTION
**Pull request by migueltylee**
Adds the type of MapEntry key and value to the VMInfoCard widget parameter 'rowKeyValues' - MapEntry<String,Widget>. It was previously not specified and thus, it was considered as dynamic by the compiler. 
Also added the implementation of PreferredSizeWidget to VMInfoCard.

**Issues fixed:**
The compiler now knows what type to expect for the key-value pairs in the VMInfoCard.

### CODE REVIEW

**Review by bkonyi — COMMENTED** [review]
Is there any reason we shouldn't just specify the types to be `MapEntry<String, Object?>` everywhere instead of requiring values to be a `Widget`?

File: packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart [hunk 1]
```diff
@@ -42,6 +42,33 @@ class VMInfoCard extends StatelessWidget {
...
+    }
+    return Size.fromHeight(
+      areaPaneHeaderHeight +
+          (rowKeyValues?.length ?? 0) * defaultRowHeight +
+          defaultSpacing,
+    );
+  }
+}
+
+MapEntry<String, Widget> stringWidgetMapEntry(
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart:59]: naming nit: maybe `_selectableTextMapEntry` instead of string widget

**Comment by kenzieschmoll** [comment]
> Is there any reason we shouldn't just specify the types to be MapEntry<String, Object?> everywhere instead of requiring values to be a Widget?

We have cases where we want the value of each entry to be a widget. If we leave this as Object, then we have to do some forking in the build methods like `if (value is Widget)` use the widget and `if (value is String)` use some SelectableText, etc. 
Using type `Widget` explicitly avoids this sort of dynamic behavior. 

One thing we could consider that would be even better than passing around Widgets would be to pass in a builder: `MapEntry<String, Widget Function()>`. This way, where the builder would be used, we'd write `entry.value()` instead of `entry.value`, and the Widget would be constructed as part of the Flutter lifecycle instead of beforehand. It is generally a bad idea in Flutter to construct widgets outside of the `build` lifecycle method and to pass them around. Building a Widget outside of the `build` method could prevent things from updating properly when things like State or BuildContext change.

File: packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart [hunk 2]
```diff
@@ -42,6 +42,33 @@ class VMInfoCard extends StatelessWidget {
...
+    if (table != null) {
+      return Size.infinite;
+    }
+    return Size.fromHeight(
+      areaPaneHeaderHeight +
+          (rowKeyValues?.length ?? 0) * defaultRowHeight +
+          defaultSpacing,
+    );
+  }
+}
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart:57]: was this intended to be part of another PR? it seems unrelated to the rest of the changes

File: packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart (hunk 1 above)
Comment (migueltylee) [reply packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart:59]: I'll change it. It still needs to be public though

File: packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart (hunk 2 above)
Comment (migueltylee) [reply packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart:57]: It will be useful to have the VmInfoCard have a size, to use it directly in the object inspector and potentially other screens in the vm tools tab. I thought I would add that in this PR, since it is also a change to the VmInfoCard.

**Comment by migueltylee** [comment]
Refactored the VmInfoCard widget to now take 'rowKeyValues' of type MapEntry<String, Widget Function(BuildContext)>, to ensure the widgets are constructed inside the VmInfoCard build method.

**Review by kenzieschmoll — APPROVED** [review]
